### PR TITLE
[untested] mkcloud: Do not rely on crowbar-backup for purge and installing packages

### PR DESCRIPTION
**Pull request by vuntz**
These steps should really not belong to the crowbar-backup script, so
import them here instead.

### CODE REVIEW

File: scripts/mkcloud [hunk 1]
```diff
@@ -1004,9 +1004,25 @@ function crowbarrestore()
         return 56
     fi
     $scp "$btarball" root@$adminip:/tmp/
-    sshrun "AGREEUNSUPPORTED=1 CB_BACKUP_IGNOREWARNING=1 bash -x /usr/sbin/crowbar-backup purge"
-    # Need to install crowbar-backup again as purge deletes it
-    sshrun "zypper --non-interactive in --no-recommends crowbar"
+
+    # Purge files to pretend we start from a clean state
+    sshrun '
+    cp -a /var/lib/crowbar/cache/etc/resolv.conf /etc/resolv.conf
```
Comment (aspiers) [inline scripts/mkcloud]: would be nicer to indent the commands inside the `sshrun`

File: scripts/mkcloud [hunk 2]
```diff
@@ -1004,9 +1004,25 @@ function crowbarrestore()
...
+
+    for service in crowbar chef-{server,solr,expander,client} couchdb apache2 named dhcpd xinetd rabbitmq-server ; do
+        [ -e /etc/init.d/$service ] && /etc/init.d/$service stop
+    done
+    killall epmd # part of rabbitmq
+    killall looper_chef_client.sh
+
+    zypper -n rm `rpm -qa|grep -e crowbar -e chef -e rubygem -e susecloud` couchdb createrepo erlang rabbitmq-server sleshammer yum-common bind bind-chrootenv dhcp-server tftp
+    rm -rf /opt/dell /etc/{bind,chef,crowbar,crowbar.install.key,dhcp3,xinetd.d/tftp} /etc/sysconfig/{dhcpd,named,rabbitmq-server} /var/lib/{chef,couchdb,crowbar,dhcp,named,rabbitmq} /var/run/{chef,crowbar,named,rabbitmq} /var/log/{apache2,chef,couchdb,crowbar,nodes,rabbitmq} /var/cache/chef /var/chef /srv/tftpboot/{discovery/pxelinux.cfg/*,nodes,validation.pem}
```
Comment (aspiers) [inline scripts/mkcloud]: Long lines alert!!  Break all these with backslashes please.

**Comment by aspiers** [comment]
Can you explain the justification for avoiding this dependency on crowbar-backup?  I'm not against it but currently it's not at all clear why you're doing it.  Also can we avoid duplicating code between the two?

**Comment by vuntz** [comment]
@aspiers the commit message says:

> These steps should really not belong to the crowbar-backup script

which really implies that I want to drop this from the crowbar-backup script :-)

**Comment by aspiers** [comment]
That still doesn't answer my (badly-phrased) first question :-)  I'll try again: **why** doesn't this code belong in the crowbar-backup script?

**Comment by vuntz** [comment]
@aspiers heh. Well, that would belong to the other PR which I'll do after https://github.com/SUSE-Cloud/cloud-tools/pull/18 gets merged. But in short, this is dangerous code (blindly removing files and packages), and we ship that. It's totally wrong to ship that.

**Comment by bmwiedemann** [comment]
The point of it being in the crowbar tool was that it has use-cases outside of mkcloud, where people want to do a fresh start with crowbar. I'd prefer to add some --I_know_what_I_am_doing option there.

**Comment by vuntz** [comment]
@bmwiedemann if people want to start fresh, they should install a new SLES. I can't see us properly supporting something where files have been wiped like that.

**Comment by vuntz** [comment]
And just for reference, installing a new SLES is how I test the backup/restore process.

**Comment by aspiers** [comment]
> But in short, this is dangerous code (blindly removing files and packages), and we ship that. It's totally wrong to ship that.

OK I can't disagree - so please include this in the commit message (and ditto for the corresponding change to cloud-tools when you submit it).

**Comment by vuntz** [comment]
@aspiers: done

**Comment by aspiers** [comment]
+1, thanks.  (BTW this review cycle would have been a lot shorter if you had written that from the start ;-)

**Comment by jdsn** [comment]
:+1:

FYI, if anybody is interested in history:
In the beginning crowbar-backup was not meant to be supported at all. Thats why it had all these hacks. And the random deletion an uninstall (aka. purge function) was way faster than deploying a new node during development of this script.

**Comment by jdsn** [comment]
:-1:
just saw that a related commit was already merged that moves this functionality to qa_crowbarsetup. The big "ssh_run" function is another indicator that this belongs there instead of in mkcloud.

**Comment by vuntz** [comment]
Rebased

**Comment by bmwiedemann** [comment]
How about splitting off an onadmin_crowbar_purge function, so that for our tests we would do backup purge restore in 3 steps instead of 2

**Comment by jdsn** [comment]
:+1:

**Comment by bmwiedemann** [comment]
my proposal is in #243

**Pull request merged**